aip/general/0203.md

```diff
@@ -162,11 +163,36 @@ the user's behalf.
 A resource with an unordered list **may** return the list in a stable order, or
 **may** return the list in a randomized, unstable order.
 
+## Rationale
+
+### Requiring field behavior
+
+The field behavior annotation explicitly states the intended behavior of the
+resource which enables better documentation. In addition, requiring the
+annotation forces the API author to explicitly consider the behavior during
+authoring of the API.
+
+Modifying field behavior after initial authoring results in
+backwards-incompatible changes in clients. For example, making an optional field
+required results in backwards-incompatible changes in the method signature of
+an RPC, or the require property for an [IaC][] client.
+
+## History
+
+In 2023-05, field behavior was made mandatory due to required and immutable
```

Comment:
Immutable is no longer mentioned above. I agree it makes sense to require it in the case that an attribute is immutable (i.e., a runtime error will be returned in the case that a client attempts to update the value) and is expected to remain immutable forever.

Reply:
Should I add something to clarify further? I can go modify the immutable section
to clarify the backwards-compatibility of each behavior in the relevant
annotation section.

Reply:
Actually I think the wording has to be updated quite a bit on immutable (it
refers specifically to requests), if it's okay I'll leave it to a different PR.

Reply:
That's fine, but I wouldn't mention immutable here, then.

Reply:
Okay I think I understand what you're saying. I've updated the wording above to
state that *all* relevant annotations must be used. I've included immutable as
an example of a field_behavior that, if added or removed, can result in a
backwards-incompatible change. Although it's arguable that any field_behavior
change can result in a backwards-incompatible client change.

Reply:
okay updated wording, PTAL.